packages/typescript-estree/src/convert.ts

```diff
@@ -1315,6 +1315,14 @@ export default function convert(config: ConvertConfig): ESTreeNode | null {
       }
 
       if (node.questionToken) {
+        if (node.questionToken.end > parameter.range[1]) {
+          parameter.range[1] = node.questionToken.end;
+          parameter.loc = getLocFor(
```

Comment:
i'm tempted to refactor ranges a little, and make some helpers for code like this